chrome/browser/net/chrome_network_delegate.cc

```diff
@@ -86,6 +87,38 @@ namespace {
 
 bool g_access_to_all_files_enabled = false;
 
+const std::regex RE_JS("\\.js(?:\\?.+)?", std::regex_constants::ECMAScript | std::regex_constants::icase);
+const std::regex RE_CSS("\\.css(?:\\?.+)?", std::regex_constants::ECMAScript | std::regex_constants::icase);
+const std::regex RE_IMAGE("\\.(?:gif|png|jpe?g|bmp|ico)(?:\\?.+)?", std::regex_constants::ECMAScript | std::regex_constants::icase);
+const std::regex RE_FONT("\\.(?:ttf|woff)(?:\\?.+)?", std::regex_constants::ECMAScript | std::regex_constants::icase);
+const std::regex RE_HTML("\\.html?(?:\\?.+)?", std::regex_constants::ECMAScript | std::regex_constants::icase);
+
+content::ResourceType detectResourceType(const std::string& url)
+{
+  if (std::regex_search(url, RE_JS))
```

Comment:
Just for a reference, I also thought about storing pairs of regexp and resource types in an array and here just iterate it, but I think in the long term the current approach is better because I expect that we will add more heuristics here.

Reply:
may be map, not array (we need to get type for regexp)? My original idea was to use chain of responsibility and each can detect & return each type. But it will add complexity and it seems that's not the case.